src/Umbraco.Web/PublishedCache/NuCache/DataSource/MsgPackContentNestedDataSerializer.cs

```diff
@@ -48,10 +53,42 @@ public ContentNestedData Deserialize(string data)
 
         public string Serialize(ContentNestedData nestedData)
         {
+            Optimize(nestedData);
+
             var bin = MessagePackSerializer.Serialize(nestedData, _options);
             return Convert.ToBase64String(bin);
         }
 
+        /// <summary>
+        /// Compress properties and map property names to shorter names
+        /// </summary>
+        /// <param name="nestedData"></param>
+        private void Optimize(ContentNestedData nestedData)
+        {
+            if (_propertyOptions.PropertyMap != null && _propertyOptions.PropertyMap.Any())
+            {
+                foreach (var map in _propertyOptions.PropertyMap)
+                {
+                    if (map.Value.compress.Equals(NucachePropertyCompressionLevel.SQLDatabase))
+                    {
+                        if (nestedData.PropertyData.TryGetValue(map.Key, out PropertyData[] properties))
+                        {
+                            foreach (var property in properties.Where(x => x.Value != null && x.Value is string))
+                            {
+                                property.Value = LZ4Pickler.Pickle(Encoding.UTF8.GetBytes(property.Value as string), _propertyOptions.LZ4CompressionLevel);
+                            }
+                        }
+                    }
+                    if (map.Value.mappedAlias != null && !map.Key.Equals(map.Value.mappedAlias)
+                        && nestedData.PropertyData.Remove(map.Key) && nestedData.PropertyData.TryGetValue(map.Key, out PropertyData[] properties2))
+                    {
+                        nestedData.PropertyData.Remove(map.Key);
```

Comment:
This 2nd Remove call isn't necessary I don't think since it's already being removed in the if statement?

Reply:
I think this alias mapping wasn't working correctly. I remember having to set the map to map to the same name in order to git it to work last time. This may be why.

Reply:
It saves a few bytes, but it might be worth removing this if block for now.

Reply:
Yep have added notes, i'll test to see how it goes.